### ISSUE TITLE
Blocking problem of the download method in Java SDK.

### ISSUE THREAD

**Issue by ztgoto** [minio/minio-java#1590]
When the file to be downloaded does not exist (404) in the Java SDK, the download method(#getObject #downloadObject) will be blocked all the time instead of throwing an exception to interrupt the download method. I don't know if there's something wrong with my usage or some other issues. I hope that the method can return quickly when the file to be downloaded does not exist, rather than being blocked all the time.

sdk version: 8.5.13

**Comment by harshavardhana** [comment]
> When the file to be downloaded does not exist (404) in the Java SDK, the download method(#getObject #downloadObject) will be blocked all the time instead of throwing an exception to interrupt the download method. I don't know if there's something wrong with my usage or some other issues. I hope that the method can return quickly when the file to be downloaded does not exist, rather than being blocked all the time.

blocked on what? please provide a reproducer and also who is your S3 provider?

**Comment by ztgoto** [comment]
> > When the file to be downloaded does not exist (404) in the Java SDK, the download method(#getObject #downloadObject) will be blocked all the time instead of throwing an exception to interrupt the download method. I don't know if there's something wrong with my usage or some other issues. I hope that the method can return quickly when the file to be downloaded does not exist, rather than being blocked all the time.
> 
> blocked on what? please provide a reproducer and also who is your S3 provider?

I use a Minio server deployed with Docker. When the Java SDK attempts to download a non - existent file, the console has already printed the HTTP 404 - related response, but the download method remains blocked all the time. The JDK 1.8 is used in my demo.

**Comment by balamurugana** [comment]
Duplicate of https://github.com/minio/minio-java/issues/1431
